plugins/link/SharedMemory.h

```diff
@@ -17,22 +17,29 @@
 #	include <string>
 #endif
 
+struct LinkedMem;
+
 class SharedMemory {
 public:
 	explicit SharedMemory();
 	~SharedMemory();
 
-	std::size_t size();
-
 	void close();
 
 	int lastError() const;
 
-	void *mapMemory(const char *name, std::size_t size);
+	bool mapMemory(const char *name);
+
+	bool isMemoryMapped();
+
+	void read(LinkedMem &dest);
```

Comment:
```suggestion
	LinkedMem read() const;
```
I think this would make the interface a bit nicer (out-params are always a bit smelly) and thanks to copy elision (return value optimization) the code should effectively be just as efficient as with the in-parameter.